main.go

```diff
@@ -17,16 +21,40 @@ func main() {
 	flag.BoolVar(&version, "version", false, "print version and exit")
 	flag.Parse()
 
+	if version {
+		fmt.Println("Browserpass host app version:", VERSION)
+		os.Exit(0)
+	}
+
+	openbsd.Pledge("stdio rpath wpath proc exec")
+
 	log.SetFormatter(&log.TextFormatter{FullTimestamp: true})
 	if verbose {
 		log.SetLevel(log.DebugLevel)
 	}
 
-	if version {
-		log.Info("Browserpass host app version: ", VERSION)
-		os.Exit(0)
+	logFilePath, err := getLogFilePath()
+	if err == nil {
+		log.Debug("Logs will being written to: ", logFilePath)
+		log.AddHook(lfshook.NewHook(logFilePath, &log.TextFormatter{FullTimestamp: true}))
+	} else {
+		log.Error("Unable to initialize logging to a file: ", err)
 	}
 
 	log.Debugf("Starting browserpass host app v%v", VERSION)
 	process()
 }
+
+func getLogFilePath() (string, error) {
+	executablePath, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+	realExecutablePath, err := filepath.EvalSymlinks(executablePath)
+	if err != nil {
+		return "", err
+	}
+	executableDir := filepath.Dir(realExecutablePath)
+	logFilePath := filepath.Join(executableDir, "browserpass.log")
```

Comment:
This seems like a strange location for the log, especially on a Linux system. In cases where the host app is installed globally, the user also won't have permissions to write the logfile there.

What's the rationale behind this choice?

Reply:
Hm, valid point, how did that not cross my mind... I chose file because I was not sure how syslog implementation would behave on Windows, and file seemed like a safe choice. However it's difficult to pick a cross-platform location for the file.

I'll try to find out now how syslog logger behaves on Windows.

Reply:
If Windows syslog is problematic, we could always use file as a fallback option on that platform.